Sources/_StringProcessing/Engine/Instruction.swift

```diff
@@ -278,6 +278,10 @@ extension Instruction {
     ///
     case backreference
 
+    case beginTypeErase
+
+    case endTypeErase
```

Comment:
What do these instructions do? What's their semantics?

Reply:
Add comments. The processor will maintain a stack of capture lists. `beginTypeErase` will push a new list onto the capture stack, so that all captures get added to that list. `endTypeErase(_: ValReg)` will convert all elements of the current capture list to an `AnyRegexOutput`, and moves it to the given value register, and pop the stack.  It's similar to how `matcher` works.

Reply:
I'm not fully parsing this. Could you write some XFAIL tests that illustrate the work that remains to be done after this PR?

Reply:
There's no XFAIL-able tests. This PR fixes the bug, but it's not as efficient as handling this in the bytecode directly. I'd be happy to chat in person also.

Reply:
How would `begin/endTypeErase` work in the context of backtracking. For that matter, how does this PR support any backtracking into `ARO` at all, i.e. isn't this forcing it to be atomic?

Reply:
I think backtracking into ARO will be supported, since `savePoints` currently stores the capture list. It can be extended to store the capture stack.